reservation_units/querysets.py

```diff
@@ -0,0 +1,418 @@
+        duration is selected as the `first_reservable_datetime`.
+
+        Variables which affect the first reservable time span and cause the ReservationUnit to be "closed":
+        - Date filter range
+        - Time filter range
+        - ReservationUnit `reservation_begins`
+        - ReservationUnit `reservation_ends`
+        - ReservationUnit `publish_ends`
+        - ApplicationRound `reservation_period_begin`
+        - ApplicationRound `reservation_period_end`
+
+        Variables which only affect the first reservable time span and allow the ReservationUnit to be "open":
+        - Minimum duration filter
+        - ReservationUnit `min_reservation_duration`
+        - ReservationUnit `max_reservation_duration`
+        - ReservationUnit `reservation_start_interval`
+        - ReservationUnit `reservations_min_days_before`
+        - ReservationUnit `reservations_max_days_before`
+        - Reservations from ReservationUnit TILAHIERARKIA TODO
+        """
+        now = timezone.localtime()
+        today = now.date()
+        two_years_from_now = today + timedelta(days=730)
+
+        #########################
+        # Default filter values #
+        #########################
+
+        if filter_date_start is None:
+            filter_date_start = today
+        if filter_date_end is None:
+            filter_date_end = two_years_from_now
+
+        if filter_time_start is not None:
+            filter_time_start = filter_time_start.replace(tzinfo=DEFAULT_TIMEZONE)
+        if filter_time_end is not None:
+            filter_time_end = filter_time_end.replace(tzinfo=DEFAULT_TIMEZONE)
+
+        ##########################
+        # Validate filter values #
+        ##########################
+
+        if filter_date_start < today:
+            raise ValueError("'reservable_date_start' must be not be in the past.")
+        elif filter_date_end < today:
+            raise ValueError("'reservable_date_end' must be not be in the past.")
+        elif filter_date_end > two_years_from_now:
+            raise ValueError("'reservable_date_end' must be not be more than two years in the future.")
+        elif filter_date_start > filter_date_end:
+            raise ValueError("'reservable_date_start' must be before 'reservable_date_end'.")
+
+        if filter_time_start is not None and filter_time_end is not None and filter_time_start >= filter_time_end:
+            raise ValueError("'reservable_time_start' must be before 'reservable_time_end'.")
+
+        if minimum_duration_minutes is not None and int(minimum_duration_minutes) < 15:
+            raise ValueError("'minimum_duration_minutes' can not be less than '15'.")
+
+        # Shortest possible reservation unit interval is 15 minutes, so it's used as the default value
+        minimum_duration_minutes = int(minimum_duration_minutes) if minimum_duration_minutes else 15
```

Comment:
Could maybe add a constant like the `two_years_from_now` for `minimum_allowed_duration_minutes = 15`. Not necessary though.